plan/logical_plan_builder.go

```diff
@@ -2204,45 +2174,42 @@ func extractTableAsNameForUpdate(p LogicalPlan, asNames map[*model.TableInfo][]*
 	}
 }
 
-func (b *planBuilder) buildDelete(delete *ast.DeleteStmt) Plan {
-	if delete.TableHints != nil {
```

Comment:
Why remove this check

Reply:
`pushTableHints` will return a `false` if the input table hints is `nil`, this check is unnecessary.